pkg/serverless/serverless.go

```diff
@@ -309,23 +306,23 @@ func WaitForNextInvocation(stopCh chan struct{}, daemon *Daemon, metricsChan cha
 }
 
 func callInvocationHandler(daemon *Daemon, arn string, deadlineMs int64, safetyBufferTimeout time.Duration, coldstart bool, invocationHandler InvocationHandler) {
+	daemon.StartInvocation()
 	timeout := computeTimeout(time.Now(), deadlineMs, safetyBufferTimeout)
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
-	doneChannel := make(chan bool)
-	go invocationHandler(doneChannel, daemon, arn, coldstart)
+	go invocationHandler(daemon, arn, coldstart)
 	select {
 	case <-ctx.Done():
 		log.Debug("Timeout detected, finishing the current invocation now to allow receiving the SHUTDOWN event")
 		daemon.FinishInvocation()
 		return
-	case <-doneChannel:
+	case <-daemon.doneChannel:
```

Comment:
Doesn't daemon.WaitForDaemon do this already? That function is already called in handleInvocation as well. You can only receive the done signal once from the channel, otherwise it will block.